## 1. Import Library & Eksplor Dataset dengan Pandas

In [246]:
# Import library awal yang dibutuhkan
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
plt.style.use("seaborn")

In [247]:
# Load dataset
data_januari = pd.read_csv("Volume Pengangkutan Sampah Di Kali Sungai Situ Waduk Bulan Januari Tahun 2021.csv",
                          encoding="latin-1")
data_januari.head()

bulan                                titik_lokasi   kecamatan   \
0      1       Kali / Sungai Sentiong - Ruas Dewi Sri  Johar Baru   
1      1         Kali / Sungai Sentiong - Ruas Abapon  Johar Baru   
2      1     Kali / Sungai Sentiong - JL Tanah Tinggi  Johar Baru   
3      1  Kali / Sungai Sentiong Jl.Galur - Pombensin  Johar Baru   
4      1           Kali / Sungai Sentiong - Jl. Timah  Johar Baru   

        wilayah   panjang/luas  satuan_panjang/luas  tanggal  \
0  Jakarta Pusat            464                   m        1   
1  Jakarta Pusat            464                   m        1   
2  Jakarta Pusat            464                   m        1   
3  Jakarta Pusat            606                   m        1   
4  Jakarta Pusat            310                   m        1   

  volume_sampah_perhari(m3)  
0                         4  
1                         4  
2                         4  
3                         4  
4                         7

In [248]:
# Melihat info dataset (tipe data, total record data, dll)
data_januari.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54343 entries, 0 to 54342
Data columns (total 8 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   bulan                      54343 non-null  int64 
 1    titik_lokasi              54343 non-null  object
 2    kecamatan                 54343 non-null  object
 3    wilayah                   54343 non-null  object
 4    panjang/luas              53475 non-null  object
 5   satuan_panjang/luas        32388 non-null  object
 6   tanggal                    54343 non-null  int64 
 7   volume_sampah_perhari(m3)  54343 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [249]:
# Menghapus jarak spasi pada kolom dataset
data_januari.columns = data_januari.columns.str.replace(" ", "")

In [250]:
# Melihat nilai statistik deskriptif dari dataset untuk keseluruhan kolom
data_januari.describe(include="all")

bulan titik_lokasi  kecamatan        wilayah panjang/luas  \
count   54343.0        54343      54343          54343        53475   
unique      NaN         1747         42              5          442   
top         NaN      PHB PLN  Kalideres  Jakarta Barat         1000   
freq        NaN           62       2914          16430         6512   
mean        1.0          NaN        NaN            NaN          NaN   
std         0.0          NaN        NaN            NaN          NaN   
min         1.0          NaN        NaN            NaN          NaN   
25%         1.0          NaN        NaN            NaN          NaN   
50%         1.0          NaN        NaN            NaN          NaN   
75%         1.0          NaN        NaN            NaN          NaN   
max         1.0          NaN        NaN            NaN          NaN   

       satuan_panjang/luas       tanggal volume_sampah_perhari(m3)  
count                32388  54343.000000                     54343  
unique                  11           NaN                        59  
top                      m           NaN                         -  
freq                 29884           NaN                     15649  
mean                   NaN     16.000000                       NaN  
std                    NaN      8.944354                       NaN  
min                    NaN      1.000000                       NaN  
25%                    NaN      8.000000                       NaN  
50%                    NaN     16.000000                       NaN  
75%                    NaN     24.000000                       NaN  
max                    NaN     31.000000                       NaN

Hasil keluaran diatas menandakan beberapa nilai-nilai dari statistik deskriptif dari kedua jenis data yakni numerik (tipe data int & float) dan kategorik (tipe data object/string) dan tentu saja setiap jenis data memiliki nilai deskriptif yang berbeda. Seperti misalnya jenis data numerik memiliki nilai deskriptif seperti count, mean, std (standard deviasi), maximum, minimum, quartil 1 (25%), quartil 2/median (50%) dan quartil 3 (75%). Sedangkan jenis data kategorik memiliki nilai deskriptif seperti count, unique, top dan freq (frekuensi data).

Sebagai contoh dapat dilihat pada penjelasan dibawah ini:

- Pada kolom titik_lokasi tidak memiliki nilai mean, standard deviasi, min, dll, tetapi memiliki nilai top, unique, freq, dll. Hal tersebut dikarenakan kolom titik_lokasi bersifat kategorik (karena memiliki tipe data string). Maka dari itu kolom titik_lokasi memiliki nilai top yakni PHB PLN, nilai freq adalah 62, dst.

- Sedangkan pada kolom tanggal tidak memiliki nilai top, freq, dll tetapi memiliki nilai mean, standard deviasi, min, max, dll. Hal tersebut dikarenakan kolom tanggal bersifat numerik. Nilai mean pada kolom tanggal adalah 16.0, nilai max adalah 31.0, nilai quartil 3 adalah 24.0, dst.

Berikut adalah kolom-kolom yang bersifat kategorik dan numerik:

- Kategorik: titik_lokasi, kecamatan, wilayah, panjang/luas, satuan_panjang/luas dan volume_sampah_perhari(m3).
- Numerik: bulan dan tanggal.

In [251]:
# Melihat tipe data dari masing-masing kolom yang ada di dalam dataframe
data_januari.dtypes

bulan                         int64
titik_lokasi                 object
kecamatan                    object
wilayah                      object
panjang/luas                 object
satuan_panjang/luas          object
tanggal                       int64
volume_sampah_perhari(m3)    object
dtype: object

In [252]:
# Melihat nama kolom yang ada di dalam dataframe
data_januari.columns

Index(['bulan', 'titik_lokasi', 'kecamatan', 'wilayah', 'panjang/luas',
       'satuan_panjang/luas', 'tanggal', 'volume_sampah_perhari(m3)'],
      dtype='object')

In [253]:
# Melihat jumlah rows dan kolom pada dataframe
data_januari.shape

(54343, 8)

In [254]:
"""
Menghitung nilai dari beberapa kolom (kecamatan, wilayah, satuan_panjang/satuan) 
dari yang sering muncul - tidak sering muncul di dalam dataframe
"""
print(data_januari["kecamatan"].value_counts())
print()

print(data_januari["wilayah"].value_counts())
print()

print(data_januari["satuan_panjang/luas"].value_counts())

Kalideres            2914
Tanjung Priok        2914
Penjaringan          2759
Cilincing            2542
Cengkareng           2480
Grogol Petamburan    2325
Cakung               2263
Kembangan            2232
Tamansari            1953
Pademangan           1891
Kebon Jeruk          1581
Tambora              1550
Kelapa Gading        1488
Koja                 1395
Palmerah             1395
Jatinegara           1364
Duren Sawit          1302
Pulo Gadung          1302
Jagakarsa            1178
Pasar Rebo           1147
Kramat Jati          1116
Makasar               961
Matraman              961
Ciracas               930
Gambir                899
Tanah Abang           899
Pancoran              868
Cilandak              837
Kebayoran Baru        837
Pasar Minggu          837
Kebayoran Lama        806
Setiabudi             775
Tebet                 744
Mampang Prapatan      713
Cipayung              651
Pesanggrahan          620
Sawah Besar           558
Senen                 527
Cempaka Puti

In [255]:
# Mengubah nama kolom satuan_panjang/luas menjadi satuan_panjang
rename_column_data = data_januari.rename(columns={"satuan_panjang/luas":"satuan_panjang"})

rename_column_data.columns

Index(['bulan', 'titik_lokasi', 'kecamatan', 'wilayah', 'panjang/luas',
       'satuan_panjang', 'tanggal', 'volume_sampah_perhari(m3)'],
      dtype='object')

In [256]:
# Memisahkan semua data yang memiliki kecamatan = Tanah Abang
data_tanah_abang = rename_column_data[rename_column_data.kecamatan == "Tanah Abang"]
data_tanah_abang.head(10)

bulan                                       titik_lokasi    kecamatan  \
88      1  Kali / Sungai BKB Ciliwung Jembatan Jl Jendral...  Tanah Abang   
89      1  Kali / Sungai BKB Ciliwung Jembatan Jl.KH Mas ...  Tanah Abang   
90      1         kali / Sungai BKB Ciliwung Pintu Air karet  Tanah Abang   
91      1              kali / Sungai BKB Ciliwung Petamburan  Tanah Abang   
92      1     kali / Sungai BKB Ciliwung Stasiun Tanah Abang  Tanah Abang   
93      1                            Kali / Sungai Cideng BI  Tanah Abang   
94      1                        Kali / Sungai Cideng kraton  Tanah Abang   
95      1      Kali / Sungai Cideng Kebon Kacang Gedung jaya  Tanah Abang   
96      1                         Kali / Sungai Cideng DEPAG  Tanah Abang   
97      1  Kali / Sungai Gendong Grand Indonesia/Pintu Ai...  Tanah Abang   

          wilayah panjang/luas satuan_panjang  tanggal  \
88  Jakarta Pusat         1500              m        1   
89  Jakarta Pusat         1000              m        1   
90  Jakarta Pusat         1000              m        1   
91  Jakarta Pusat         1500              m        1   
92  Jakarta Pusat         1000              m        1   
93  Jakarta Pusat         1000              m        1   
94  Jakarta Pusat         1000              m        1   
95  Jakarta Pusat         1000              m        1   
96  Jakarta Pusat         1000              m        1   
97  Jakarta Pusat         1000              m        1   

   volume_sampah_perhari(m3)  
88                      4.48  
89                      4.48  
90                      4.48  
91                         4  
92                         0  
93                         1  
94                      2.24  
95                      4.48  
96                      2.24  
97                         1

In [257]:
# Memisahkan semua data yang memiliki wilayah = Jakarta Timur
data_jaktim = rename_column_data[rename_column_data.wilayah == "Jakarta Timur"]
data_jaktim.head()

bulan                                       titik_lokasi kecamatan  \
1366      1  Kali Cipinang Jembatan Radar AURI - Jembatan M...   Ciracas   
1367      1  Kali Cipinang Jembatan Mekarsari - Jembatan Tipar   Ciracas   
1368      1  Kali Cipinang - Jembatan Mekarsari - Jembatan ...   Ciracas   
1369      1  Kali Cipinang Belakang SMA 99 -Jembatan Peti Mati   Ciracas   
1370      1  Kali Cipinang Jembatan Peti Mati - Jembatan Pa...   Ciracas   

            wilayah panjang/luas satuan_panjang  tanggal  \
1366  Jakarta Timur         1800              m        1   
1367  Jakarta Timur         1200              m        1   
1368  Jakarta Timur         1000              m        1   
1369  Jakarta Timur         2000              m        1   
1370  Jakarta Timur         1000              m        1   

     volume_sampah_perhari(m3)  
1366                         -  
1367                         -  
1368                         -  
1369                         -  
1370                         -

In [258]:
# Memisahkan semua data yang memiliki satuan_panjang = Ha
data_Ha = rename_column_data[rename_column_data.satuan_panjang == "Ha"]
data_Ha.head(10)

bulan                              titik_lokasi      kecamatan  \
177      1                         PHB Akses Marunda      Cilincing   
234      1                             Waduk Kodamar  Kelapa Gading   
346      1           Waduk Sunter Selatan SPBU Shell  Tanjung Priok   
347      1           Waduk Sunter Selatan Pos Polisi  Tanjung Priok   
348      1       Waduk Sunter Selatan SPBU Pertamina  Tanjung Priok   
349      1                  Waduk Sunter Utara Bisma  Tanjung Priok   
350      1            Waduk Sunter Utara Stadion BMW  Tanjung Priok   
352      1                 Waduk Sunter Utara Daikin  Tanjung Priok   
400      1              Waduk Sunter Timur depan Gor  Tanjung Priok   
401      1  Waduk Sunter Timur depan Astra Container  Tanjung Priok   

           wilayah panjang/luas satuan_panjang  tanggal  \
177  Jakarta Utara            1             Ha        1   
234  Jakarta Utara            8             Ha        1   
346  Jakarta Utara           10             Ha        1   
347  Jakarta Utara            9             Ha        1   
348  Jakarta Utara            9             Ha        1   
349  Jakarta Utara           10             Ha        1   
350  Jakarta Utara           10             Ha        1   
352  Jakarta Utara           10             Ha        1   
400  Jakarta Utara          6.5             Ha        1   
401  Jakarta Utara          6.5             Ha        1   

    volume_sampah_perhari(m3)  
177                         0  
234                         0  
346                         0  
347                         0  
348                         0  
349                         3  
350                         3  
352                         3  
400                         0  
401                         0

## 2. Data Preprocessing

In [259]:
# Mengecek nilai kosong (NaN) dari masing-masing kolom di dataframe
data_januari.isnull().sum()

bulan                            0
titik_lokasi                     0
kecamatan                        0
wilayah                          0
panjang/luas                   868
satuan_panjang/luas          21955
tanggal                          0
volume_sampah_perhari(m3)        0
dtype: int64

In [260]:
# Mengecek apakah di dalam dataframe terdapat nilai duplikat/tidak
data_januari.duplicated().sum()

0

In [261]:
# Mengubah format penamaan kolom dari yang semula huruf kecil menjadi huruf gabungan (huruf besar & kecil)
data_januari = data_januari.rename(str.title, axis="columns")

In [262]:
# Menghapus karakter yang berada di kolom "Panjang/Luas"
data_januari["Panjang/Luas"] = data_januari["Panjang/Luas"].str.replace(r"\D", "")

In [263]:
# Mengahapus karakter "-" di kolom "Volume_Sampah_Perhari(M3)" dengan nilai NaN
data_januari["Volume_Sampah_Perhari(M3)"] = data_januari["Volume_Sampah_Perhari(M3)"].replace("-", np.nan)

Alasan saya mengganti icon "-" dengan nilai NaN adalah supaya dapat memudahkan ketika dalam mengganti data karena nilai NaN tersebut akan saya isi dengan nilai yang ada di statistik deskriptif yaitu nilai mean.

In [264]:
# Mengubah tipe data di kolom "Panjang/Luas" dari string menjadi float
data_januari[["Panjang/Luas"]] = data_januari[["Panjang/Luas"]].apply(pd.to_numeric)

In [265]:
data_januari["Panjang/Luas"].value_counts()

1000.0    6512
500.0     5735
2000.0    2883
1500.0    2294
800.0     2108
          ... 
1350.0       1
1820.0       1
1360.0       1
2700.0       1
1320.0       1
Name: Panjang/Luas, Length: 412, dtype: int64

In [266]:
# Mengubah tipe data pada kolom "Volume_Sampah_Perhari(M3)" dari string menjadi float
data_januari["Volume_Sampah_Perhari(M3)"] = pd.to_numeric(data_januari["Volume_Sampah_Perhari(M3)"], errors="coerce")

data_januari.tail()

Bulan                                       Titik_Lokasi Kecamatan  \
54338      1  Kali Cakung Drain (Jembatan Tinggi-Perbatasan ...    Cakung   
54339      1                                   PHB Rawa Teratai    Cakung   
54340      1                                 PHB Rawa Gelam III    Cakung   
54341      1                                   PHB Metropos KIP    Cakung   
54342      1                                 PHB Warung Jengkol    Cakung   

             Wilayah  Panjang/Luas Satuan_Panjang/Luas  Tanggal  \
54338  Jakarta Timur         800.0                   m       31   
54339  Jakarta Timur         700.0                   m       31   
54340  Jakarta Timur        1000.0                   m       31   
54341  Jakarta Timur        1000.0                   m       31   
54342  Jakarta Timur         400.0                   m       31   

       Volume_Sampah_Perhari(M3)  
54338                        NaN  
54339                        NaN  
54340                        NaN  
54341                        NaN  
54342                        2.0

In [267]:
"""
Mengubah tipe data pada dari beberapa kolom (kolom "Kecamatan", "Wilayah", dan "Satuan_Panjang/Luas") 
menjadi int (semula adalah string)

from sklearn.preprocessing import LabelEncoder

labelencoder = LabelEncoder()

data_januari["Kecamatan"] = labelencoder.fit_transform(data_januari["Kecamatan"])
data_januari["Wilayah"] = labelencoder.fit_transform(data_januari["Wilayah"])
data_januari["Satuan_Panjang/Luas"] = labelencoder.fit_transform(data_januari["Satuan_Panjang/Luas"])

data_januari.tail()
"""

'\nMengubah tipe data pada dari beberapa kolom (kolom "Kecamatan", "Wilayah", dan "Satuan_Panjang/Luas") \nmenjadi int (semula adalah string)\n\nfrom sklearn.preprocessing import LabelEncoder\n\nlabelencoder = LabelEncoder()\n\ndata_januari["Kecamatan"] = labelencoder.fit_transform(data_januari["Kecamatan"])\ndata_januari["Wilayah"] = labelencoder.fit_transform(data_januari["Wilayah"])\ndata_januari["Satuan_Panjang/Luas"] = labelencoder.fit_transform(data_januari["Satuan_Panjang/Luas"])\n\ndata_januari.tail()\n'

In [ ]:
# Mengganti nilai NaN yang ada disetiap kolom di data_januari dengan nilai mean
mean = data_januari.mean()
data_januari = data_januari.replace(np.nan, mean)

data_januari.isnull().sum()

Karena pada kolom "Satuan_Panjang/Luas" terdapat 21955 nilai data yang kosong maka saya akan menghapus kolom "Satuan_Panjang/Luas". Selain karena memiliki nilai NaN yang sangat banyak, kolom "Satuan_Panjang/Luas" sudah tidak dibutuhkan dalam mencari prediksi terkait tingkat pengangkutan volume sampah.

Selain kolom tersebut saya juga akan menghapus kolom "Bulan" karena sudah jelas dari data ini hanya berisi volume sampah yang berhasil diangkut di Bulan Januari tahun 2021.

In [ ]:
# Menghapus kolom yang sudah tidak dibutuhkan yaitu kolom "Satuan_Panjang/Luas" dan kolom "Bulan"
data_januari = data_januari.drop(["Satuan_Panjang/Luas", "Bulan"], axis=1)

In [ ]:
# Mengecek nilai outliers untuk kolom "Panjang/Luas" dengan nilai quartil dan IQR
Q1 = data_januari["Panjang/Luas"].quantile(0.25)
Q3 = data_januari["Panjang/Luas"].quantile(0.75)

# Menghitung interquartile range maka rumusnya adalah quartil 3 (75%) - quartil 1 (25%)
IQR = Q3 - Q1

# Melihat ukuran (baris dan kolom) sebelum data yang outliers dibuang
print("Ukuran Data dari Kolom Panjang/Luas Sebelum Nilai Outliers Dibuang: ", data_januari.shape)

# Menghapus nilai outlier dari kolom "Panjang/Luas"
data_januari = data_januari[~((data_januari["Panjang/Luas"] < (Q1 - 1.5 * IQR)) | 
                              (data_januari["Panjang/Luas"] > (Q3 + 1.5 * IQR)))]

# Melihat ukuran (baris dan kolom) setelah data outliers dibuang dari kolom "Panjang/Luas"
print("Ukuran Data dari Kolom Panjang/Luas Setelah Nilai Outliers Dibuang: ", data_januari.shape)

Berdasarkan hasil output diatas terdapat 10.540 data outlier dari kolom "Panjang/Luas" dan jika dihapus akan mengkhawatirkan distribusi data di kolom Panjang/Luas tidak balanced/tidak seimbang.

In [ ]:
# Mengubah tipe data pada kolom "Panjang/Luas" dan "Volume_Sampah_Perhari(M3)" dari float menjadi integer
data_januari["Panjang/Luas"] = data_januari["Panjang/Luas"].astype(int)
data_januari["Volume_Sampah_Perhari(M3)"] = data_januari["Volume_Sampah_Perhari(M3)"].astype(int)

data_januari.info()

## 3. Data Visualisasi serta Analisa per Grafik

In [ ]:
# Melihat perbandingan antar wilayah di Provinsi DKI Jakarta (berdasarkan volume sampah per hari)
plt.figure(figsize=(25, 20))

plt.subplot(4, 3, 1)
sns.barplot(x="Wilayah", y="Volume_Sampah_Perhari(M3)", data=data_januari)
plt.title("Grafik Wilayah Jakarta Berdasarkan \n Volume Sampah Per Hari di Bulan Januari 2021", 
          fontsize=13, pad=13)

plt.subplot(4, 3, 2)
sns.countplot(x="Wilayah", data=data_januari)
plt.title("Grafik Wilayah Jakarta di Bulan Januari 2021", fontsize=13, pad=13)
plt.show()

Berdasarkan grafik diagram diatas maka dapat diambil beberapa kesimpulan diantaranya:

1. Wilayah Jakarta Pusat memiliki volume sampah yang paling tinggi diantara wilayah yang lain sedangkan Wilayah Jakarta Utara memiliki volume sampah yang sangat sedikit. Jika diurutkan dalam hal volume sampah yang berhasil diangkut maka akan terjadi susunan seperti dibawah ini:
- Jakarta Pusat.
- Jakarta Barat.
- Jakarta Selatan.
- Jakarta Timur.
- Jakarta Utara.

2. Grafik kedua merupakan grafik yang memperlihatkan perbandingan Wilayah yang ada di Provinsi DKI Jakarta yang dihitung per Kecamatan tanpa ada keterkaitan dalam pengangkutan sampah per harinya. Wilayah Jakarta Barat memiliki nilai yang sangat tinggi yakni mencapai 16.000 sedangkan Wilayah Jakarta Utara memiliki nilai perhitungan yang sangat kecil yakni hanya mencapai kurang lebih 3.000 saja. Jika diurutkan maka hasilnya adalah sebagai berikut:
- Jakarta Barat.
- Jakarta Timur.
- Jakarta Selatan.
- Jakarta Pusat.
- Jakarta Utara.

In [ ]:
# Melihat perbandingan antar kecamatan dalam hal pengangkutan sampah yang ada di Provinsi DKI Jakarta
plt.figure(figsize=(15, 10))
plt.xticks(rotation=90)

sns.countplot(x="Kecamatan", data=data_januari)
plt.title("Perbandingan Kecamatan Yang Paling Sering Mengangkut Sampah Sungai Pada Bulan Januari 2021", fontsize=13, pad=13)
plt.show()

Berdasarkan grafik diatas terkait perbandingan jumlah perhitungan total antar Kecamatan yang didasari oleh kolom Titik_Lokasi dalam menghasilkan/mengumpulkan volume sampah per harinya di Provinsi DKI Jakarta, maka dapat diambil kesimpulan sebagai berikut:

1. Kecamatan yang mempunyai nilai tertinggi adalah Kecamatan Kalideres yang memiliki nilai lebih dari 2.500. Jumlah nilai tersebut tidak ada keterkaitannya dengan volume sampah per harinya melainkan didasarkan atas Titik_Lokasi dari sampah-sampah yang berhasil diangkut per harinya.
2. Kecamatan seperti Kembangan, Cengkareng, Kebon Jeruk, Grogol dan Palmerah yang saya kira memiliki nilai tertinggi ternyata masih kalah jauh dari Kecamatan Kalideres (Hanya Kecamatan Cengkareng dan Grogol yang memiliki perbandingan nilai yang sangat kecil dari Kecamatan Kalideres).
3. Kecamatan yang memiliki titik lokasi terendah dalam hal pengangkutan sampah di Provinsi DKI Jakarta adalah Kecamatan Cilincing yang berada di Wilayah Jakarta Utara yang memiliki titik lokasi pengangkutan hanya 186 saja. Kalah tipis dari Kecamatan Koja yang berada di Wilayah yang sama (Jakarta Utara).

Harus diingat bahwa perhitungan dalam memilih Kecamatan tertinggi sampai terendah didasarkan bukan dari volume sampah yang berhasil dingkut per harinya melainkan dari pengumpulan sampah di suatu titik lokasi tertentu saja seperti, kali, saluran penghubung (PHB), waduk, dll.

In [ ]:
# Melihat distribusi dan kepadatan pada kolom "Umur" dengan menggunakan teknik kernel density estimate (KDE)
plt.figure(figsize=(10, 5))
    
sns.lineplot(x="Tanggal", y="Volume_Sampah_Perhari(M3)", color="green", data=data_januari)
plt.title("Pendistribusian/Kepadatan Data Berdasarkan Umur Penduduk di Desa Sukabumi Selatan", fontsize=13, pad=13)
plt.show()

Dari grafik garis (Line plot) diatas menunjukkan intensitas pengangkutan sampah yang dilakukan oleh Dinas Lingkungan Hidup Provinsi DKI Jakarta di setiap hari pada bulan Januari tahun 2021. Hasilnya adalah cenderung menunjukkan terjadinya trend kenaikan volume sampah di setiap harinya dan puncaknya terjadi di tanggal 18 Januari 2021. Namun pada tanggal 25 Januari terjadi penurunan trend dalam mengangkut sampah di berbagai titik lokasi yang telah ditentukan. Tetapi walaupun begitu penurunan trend data ini tidak terlalu mengkhawatirkan sebab setelah tanggal 25 Januari, tepatnya pada tanggal 26 Januari, terjadi kenaikan trend yang sangat signifikan walaupun di hari esoknya (27 Januari) juga terjadi penurunan trend kembali dalam mengangkut sampah di beberapa titik lokasi yang telah ditentukan (Walaupun tidak sedrastis tanggal 25).

In [ ]:
# Melihat korelasi antar kolom yang memiliki tipe data numerik
plt.figure(figsize=(10, 5))

korelasi_data = sns.heatmap(data_januari.corr(), cmap="GnBu", annot=True)
korelasi_data.set_title("Melihat Korelasi Antar Kolom Data yang Bersifat Numerik/Angka", 
                        fontdict={"fontsize":13}, pad=13);

In [ ]:
# Melihat pola hubungan antara 2 kolom mennggunakan scatter plot
plt.title("Melihat Persebaran Data Antar 2 Kolom")
sns.scatterplot(x="Tanggal", y="Volume_Sampah_Perhari(M3)", hue="Wilayah", data=data_januari)

In [ ]:
"""
# Melihat presentase masing-masing wilayah Provinsi DKI Jakarta berdasarkan volume sampah disetiap harinya
data_pie = data_januari.groupby("Wilayah")["Volume_Sampah_Perhari(M3)"].sum().reset_index()
plt.figure(figsize=(10, 8))
#colors = ["blue", "red", "green", "yellow", "purple"]

plt.pie(data_pie["Volume_Sampah_Perhari(M3)"], labels = data_pie["Wilayah"], autopct = "%1.2f%%")
plt.title("Melihat Presentase Antar Wilayah Berdasarkan Volume Sampah Dari Sungai Provinsi DKI Jakarta", pad=13, fontsize=13)
plt.show()
"""